### PR TITLE
allow more online models to load tokens file from the memory

### PR DESCRIPTION
**Pull request by shawl336**
for online paraformer and ctc, their symbol tables are able to be initialized from tokens in a memory buffer

### CODE REVIEW

**Comment by shawl336** [comment]
for the keywords spotter example, the kws model failed to spot any keywords among the given test wavs. however, if i record a wav myself, only the @你好军哥 can be spotted, could u help check the keywords-spotter-buffered-tokens-keywords-c-api.c, I can not figure the problem, other than imputing it to the model accuray

**Comment by shawl336** [comment]
the check "test-nodejs / test-nodejs (ubuntu-latest, 3.8) (pull_request)" failed due to the wasm filesystem, both keywords_file and keywords_buf are provided, which did not pass the validate().
should i add a special case for wasm in the validate or change the test code for the wasm filesystem (i am not familiar to wasm)

File: .github/workflows/c-api-from-buffer.yaml [hunk 1]
```diff
@@ -120,3 +120,99 @@ jobs:
...
+          rm sherpa-onnx-streaming-paraformer-bilingual-zh-en.tar.bz2
+
+          ls -lh sherpa-onnx-streaming-paraformer-bilingual-zh-en
+          echo "---"
+          ls -lh sherpa-onnx-streaming-paraformer-bilingual-zh-en/test_wavs
+
+          export LD_LIBRARY_PATH=$PWD/build/install/lib:$LD_LIBRARY_PATH
+          export DYLD_LIBRARY_PATH=$PWD/build/install/lib:$DYLD_LIBRARY_PATH
+
+          ./streaming-paraformer-buffered-tokens-hotwords-c-api
```
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]: By the way, we don't support hotwords when paraformer is used.

Only transducer models can support hotwords.

**Comment by csukuangfj** [comment]
> for the keywords spotter example, the kws model failed to spot any keywords among the given test wavs

Please check the keywords contained in the keywords.txt and check whether the test wave file contains a keyword.

File: .github/workflows/c-api-from-buffer.yaml (hunk 1 above)
Comment (shawl336) [reply .github/workflows/c-api-from-buffer.yaml]: thx for the noting, I didnot use hotwords for the paraformer, this is a copy-issue

**Comment by shawl336** [comment]
> > for the keywords spotter example, the kws model failed to spot any keywords among the given test wavs
> 
> Please check the keywords contained in the keywords.txt and check whether the test wave file contains a keyword.

my mistake, I confused with the test_wavs/test_keywords.txt and the keywords.txt, now the keywords can be spotted out

**Comment by csukuangfj** [comment]
> the check "test-nodejs / test-nodejs (ubuntu-latest, 3.8) (pull_request)" failed due to the wasm filesystem, both keywords_file and keywords_buf are provided, which did not pass the validate(). should i add a special case for wasm in the validate or change the test code for the wasm filesystem (i am not familiar to wasm)


> should i add a special case for wasm in the validate or change the test code for the wasm filesystem 

Please focus only on the C++ and C APIs. I will fix the errors related to other programming language APIs.

File: .github/workflows/c-api-from-buffer.yaml [hunk 2]
```diff
@@ -120,3 +120,99 @@ jobs:
           ./streaming-zipformer-buffered-tokens-hotwords-c-api
 
           rm -rf sherpa-onnx-streaming-zipformer-*
+
+      - name: Test streaming paraformer with tokens and hotwords loaded from buffers
+        shell: bash
+        run: |
+          gcc -o streaming-paraformer-buffered-tokens-hotwords-c-api ./c-api-examples/streaming-paraformer-buffered-tokens-hotwords-c-api.c \
```
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]:
```suggestion
          gcc -o streaming-paraformer-buffered-tokens-hotwords-c-api ./c-api-examples/streaming-paraformer-buffered-tokens-c-api.c \
```

File: .github/workflows/c-api-from-buffer.yaml [hunk 3]
```diff
@@ -120,3 +120,99 @@ jobs:
...
+            -I ./build/install/include \
+            -L ./build/install/lib/ \
+            -l sherpa-onnx-c-api \
+            -l onnxruntime
+
+          ls -lh streaming-paraformer-buffered-tokens-hotwords-c-api
```
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]:
```suggestion
          ls -lh streaming-paraformer-buffered-tokens-c-api
```

File: .github/workflows/c-api-from-buffer.yaml [hunk 4]
```diff
@@ -120,3 +120,99 @@ jobs:
...
+
+          if [[ ${{ matrix.os }} == ubuntu-latest ]]; then
+            ldd ./streaming-paraformer-buffered-tokens-hotwords-c-api
```
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]:
```suggestion
            ldd ./streaming-paraformer-buffered-tokens-c-api
```

File: .github/workflows/c-api-from-buffer.yaml [hunk 5]
```diff
@@ -120,3 +120,99 @@ jobs:
...
+            echo "----"
+            readelf -d ./streaming-paraformer-buffered-tokens-hotwords-c-api
```
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]:
```suggestion
            readelf -d ./streaming-paraformer-buffered-tokens-c-api
```

File: .github/workflows/c-api-from-buffer.yaml (hunk 1 above)
Comment (csukuangfj) [inline .github/workflows/c-api-from-buffer.yaml]:
```suggestion
          ./streaming-paraformer-buffered-tokens-c-api
```

**Review by csukuangfj — COMMENTED** [review]
Thanks! Looks good to me. Only left some minor comments.

**Comment by shawl336** [comment]
it is ready to merge for me

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Pull request merged**